Watchman.Discord/Areas/Help/Controllers/HelpController.cs

```diff
@@ -1,38 +1,68 @@
 using Devscord.DiscordFramework.Framework.Commands.Parsing.Models;
 using Devscord.DiscordFramework.Middlewares.Contexts;
 using Devscord.DiscordFramework.Services.Factories;
-using System;
-using System.Collections.Generic;
+using System.Linq;
 using System.Text;
+using Newtonsoft.Json;
+using Watchman.Common.Strings;
 using Watchman.Cqrs;
 using Watchman.DomainModel.Help.Queries;
-using Watchman.Integrations.MongoDB;
+using Devscord.DiscordFramework.Framework.Commands.Responses;
 
 namespace Watchman.Discord.Areas.Help.Controllers
 {
-    public class HelpController
+    public class HelpController : IController
     {
         private readonly IQueryBus _queryBus;
-        private readonly ICommandBus _commandBus;
-        private readonly MessagesServiceFactory messagesServiceFactory;
-        private readonly ISession _session;
+        private readonly MessagesServiceFactory _messagesServiceFactory;
 
-        public HelpController(IQueryBus queryBus, ICommandBus commandBus, ISessionFactory sessionFactory, MessagesServiceFactory messagesServiceFactory)
+        public HelpController(IQueryBus queryBus, MessagesServiceFactory messagesServiceFactory)
         {
             this._queryBus = queryBus;
-            this._commandBus = commandBus;
-            this.messagesServiceFactory = messagesServiceFactory;
-            this._session = sessionFactory.Create();
+            this._messagesServiceFactory = messagesServiceFactory;
         }
 
         [DiscordCommand("help")]
         public void PrintHelp(DiscordRequest request, Contexts contexts)
         {
-            var result = this._queryBus.Execute(new GetHelpMessageQuery(this._session));
-            var messagesService = messagesServiceFactory.Create(contexts);
-            messagesService.SendMessage(result.HelpMessage);
+            if (request.Arguments.Any() && request.Arguments.First().Values.Any(x => x == "json"))
+            { 
+                PrintJsonHelp(request, contexts);
+                return;
+            }
+            
+            var result = this._queryBus.Execute(new GetHelpInformationQuery(contexts.Server.Id));
+
+            var lines = new List<string>();
+            foreach (var helpInfo in result.HelpInformations)
+            {
+                var line = new StringBuilder("-" + helpInfo.Names.Aggregate((x, y) => x + " / -" + y));
+
+                line.Append(" => ");
+                line.Append(helpInfo.Descriptions.First(x => x.Name == helpInfo.DefaultDescriptionName).Details);
+                lines.Add(line.ToString());
+            }
+
+            var messageBuilder = new StringBuilder().PrintManyLines(lines.ToArray());
+            var messagesService = _messagesServiceFactory.Create(contexts);
+            messagesService.SendResponse(x => x.PrintHelp(messageBuilder.ToString()), contexts);
+        }
+
+        public void PrintJsonHelp(DiscordRequest request, Contexts contexts)
```

Comment:
Do tego powinien być serwis HelpService w warstwie serwisów

Wrzuć tam GenerateJsonHelp() i zrób GenerateHelp() => to co sie generuje w PrintHelp()